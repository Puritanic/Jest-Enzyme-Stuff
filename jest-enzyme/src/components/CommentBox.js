import React from 'react';

class CommentBox extends React.Component {
	state = {
		comment: '',
	};

	onChange = e => this.setState({ comment: e.target.value });

	onSubmit = () => {
		const comment = {
			id: this.state.comment.length * Math.random() * 10,
			content: this.state.comment,
			author: 'Anon',
		};
		this.props.saveComment(comment);
		this.setState({ comment: '' });
	};

	render() {
		return (
			<div className="comment__box">
				<textarea
					onChange={this.onChange}
					value={this.state.comment}
					name="comment_field"
					className="comment_field"
					cols="30"
					rows="10"
				/>
				<button className="comment__button" onClick={this.onSubmit}>
					Submit
				</button>
			</div>
		);
	}
}

export default CommentBox;
