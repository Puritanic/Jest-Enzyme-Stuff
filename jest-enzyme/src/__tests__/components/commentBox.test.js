/* eslint-disable no-undef */
import React from 'react';

import CommentBox from '../../components/CommentBox';

describe('<CommentBox />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<CommentBox />);
	});

	it('should render', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should have textarea', () => {
		expect(wrapper.find('textarea').length).toBe(1);
	});

	it('should have button', () => {
		expect(wrapper.find('button').length).toBe(1);
	});

	it('should have correct class name', () => {
		expect(wrapper.hasClass('comment__box')).toBe(true);
	});

	describe('Input group test', () => {
		beforeEach(() => {
			wrapper.find('.comment_field').simulate('change', { target: { value: 'new comment' } });
		});

		it('should show entered text', () => {
			expect(wrapper.state().comment).toBe('new comment');
		});

		it('should clear input after submit ', () => {
			wrapper.setProps({ saveComment: data => data });
			wrapper.find('.comment__button').simulate('click');
			expect(wrapper.state().comment).toBe('');
		});

		it('should call saveComment on submit', () => {
			const saveComment = jest.fn();
			wrapper.setProps({ saveComment });
			wrapper.find('.comment__button').simulate('click');
			expect(saveComment).toBeCalled();
		});
	});
});
