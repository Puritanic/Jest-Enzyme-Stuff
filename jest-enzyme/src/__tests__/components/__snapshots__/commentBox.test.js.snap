// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentBox /> should render 1`] = `
<div
  className="comment__box"
>
  <textarea
    className="comment_field"
    cols="30"
    name="comment_field"
    onChange={[Function]}
    rows="10"
  />
  <button
    className="comment__button"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
